app/javascript/modCenter/singleArticle/index.jsx

```diff
@@ -31,16 +31,22 @@ export default class SingleArticle extends Component {
     const { articleOpened } = this.state;
     const { id, title, publishedAt, cachedTagList, user, key } = this.props;
     const tags = cachedTagList.split(', ').map((tag) => {
-      return (
-        <span className="mod-article-tag fs-s ff-accent lh-base" key={key}>
-          <span className="article-hash-tag">#</span>
-          {tag}
-        </span>
-      );
+      if (tag) {
```

Comment:
The `if (tag) {` shouldn't be necessary. If we're mapping over an items, it will exist.

Reply:
Right Nick, I agree. The issue was, with an empty `tag` list, the function still returned a solitary `#` on account of this span in the `return` block: `<span className="article-hash-tag">#</span>`

If there is a more elegant solution, I would like to implement it 😄

Reply:
Yeah, not sure how to work around this either without having the `if (tag)` 🤔

Reply:
Yeah, not sure how to work around this either without having the `if (tag)` 🤔

Reply:
Ah okay, I figured out why we should stick with `if (tag)`.  If an article has no tags, `article.cachedTagList` will equal `['']`. Therefore, checking `if (tag)` seems like the most performant way to check the validity of that tag.

Reply:
`article.cachedTagList` is a string though, so if you check if it's an empty string you can skip the string split/mapping altogether. I made it a ternary but it doesn't have to be.

```jsx
    const tags = !cachedTagList ? null : cachedTagList.split(', ').map((tag) => {
        return (
          <span className="mod-article-tag fs-s ff-accent lh-base" key={key}>
            <span className="article-hash-tag">#</span>
            {tag}
          </span>
        );
    });
```

Reply:
Ohhh smart. Will update!